### DISCUSSION TITLE
Template Variables Cannot Be Used on Data Mapping

### DISCUSSION THREAD

**Discussion by cpmoore** [grafana/grafana#38401] [Dashboards, Panels & Visualizations]
I'm trying to setup a repeating panel to monitor if a list of processes are active or not.
When the process isn't active a document is not send to elasticsearch therefore the value is null
I'm trying to use data mapping to convert that null value into text saying `[[subsystem]] NOT ACTIVE`
However it doesn't appear the template variables can be injected into data mapping using either the legacy data mapping or the new "No Value" option in the Stat panel

**What happened**:
When the value for a stat panel is null, template variables are not injected into the display value for the `No Value` option

Configuration
![image](https://user-images.githubusercontent.com/22562764/88681892-a855a880-d0b7-11ea-95b9-61d2325bfb81.png)

Stat panel has a value
![image](https://user-images.githubusercontent.com/22562764/88681915-aee42000-d0b7-11ea-8b6c-bc168a2685f2.png)

Stat panel value is null
![image](https://user-images.githubusercontent.com/22562764/88681939-b6a3c480-d0b7-11ea-9aef-f71ffa44c661.png)

**What you expected to happen**:
Template values would be injected into the `No Value` configuration
Instead of saying `[[subsystem]] NOT ACTIVE` The `[[subsystem]]` bit will be replaced with the subsystem variable to become `CIF_SERVER NOT ACTIVE`

**How to reproduce it (as minimally and precisely as possible)**:
Create a variable for the dashboard
![image](https://user-images.githubusercontent.com/22562764/88688375-df7b8800-d0be-11ea-99d3-eb0fe787d713.png)

Add any query that returns "No data"
![image](https://user-images.githubusercontent.com/22562764/88689167-c8896580-d0bf-11ea-8bfe-aed004b95f50.png)

Update the No Value configuration in the Stat panel to use your template variable
![image](https://user-images.githubusercontent.com/22562764/88689264-e5259d80-d0bf-11ea-925f-c7c2868eb8b6.png)

**Anything else we need to know?**:
I attempted with the $variable name formatting also with the same results.
As well as the legacy data mapping for null -> [[subsystem]] 

**Environment**:
- Grafana version: 7.1.1
- Data source type & version: Elasticsearch 7
- OS Grafana is installed on: Windows
- User OS & Browser: Chrome

**Comment by aknuds1** [comment]
Sounds as if this is a support request? Please direct questions to our community [forums](https://community.grafana.com) or [Slack](http://slack.raintank.io). GitHub issues are for bugs and feature requests. Please let me know if I misunderstood.

**Comment by cpmoore** [comment]
How is this not a bug?
The template variable should be injected into the data mapping and it isn’t. 

The metric for the panel has the null value, not the template variable.

**Comment by aknuds1** [comment]
It's far from clear given how you've written the issue description. Can you please rewrite it using our standard bug template, describing for example how to reproduce the issue and what differs from your expectation?

**Comment by cpmoore** [comment]
Is that clearer?

**Comment by aknuds1** [comment]
Thanks, much better.

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
